http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -283,6 +286,15 @@ class LeonardoService(
       _ <- checkClusterPermission(userInfo, GetClusterStatus, cluster) //throws 404 if no auth
     } yield cluster
 
+  // throws 404 if nonexistent or no permissions
+  def getClusterAPI(userInfo: UserInfo, googleProject: GoogleProject, clusterName: ClusterName)(
+    implicit ev: ApplicativeAsk[IO, TraceId]
+  ): IO[GetClusterResponse] =
+    for {
+      cluster <- getActiveClusterDetails(userInfo, googleProject, clusterName) //throws 404 if nonexistent
+      runtimeConfig <- Queries.getRuntime(cluster.runtimeConfigId).transaction
```

Comment:
It's a bit of a shame to get the cluster + runtime in 2 transactions… is there a way we could run 1 DB query for this information, in 1 transaction?

Reply:
added https://github.com/DataBiosphere/leonardo/pull/1238/files#diff-a96db23bf811a25fd81e8e4c55710030